src/app/features/import/import-column-mapping/import-column-mapping.component.ts

```diff
@@ -0,0 +1,125 @@
+import { ColumnMapping } from "../column-mapping";
+import { ComponentType } from "@angular/cdk/overlay";
+import { EntityRegistry } from "../../../core/entity/database-entity.decorator";
+import { EntityConstructor } from "../../../core/entity/model/entity";
+import { MatDialog } from "@angular/material/dialog";
+import { ImportService } from "../import.service";
+
+/**
+ * Import sub-step: Let user map columns from import data to entity properties
+ * and define value matching and transformations.
+ */
+@Component({
+  selector: "app-import-column-mapping",
+  templateUrl: "./import-column-mapping.component.html",
+  styleUrls: ["./import-column-mapping.component.scss"],
+})
+export class ImportColumnMappingComponent implements OnChanges {
+  @Input() rawData: any[] = [];
+  @Input() columnMapping: ColumnMapping[] = [];
+  @Output() columnMappingChange = new EventEmitter<ColumnMapping[]>();
+
+  @Input() set entityType(value: string) {
+    this.entityCtr = this.entities.get(value);
+    this.mappingCmp = {};
+    this.allProps = [...this.entityCtr.schema.entries()]
+      .filter(([_, schema]) => schema.label)
+      .map(([name, schema]) => {
+        const cmp = this.importService.getMappingComponent(schema);
+        if (cmp) {
+          this.mappingCmp[name] = cmp;
+        }
+        return name;
+      });
+  }
+
+  private entityCtr: EntityConstructor;
+  // entity properties that have a label
+  allProps: string[] = [];
+  // properties that need further adjustments through a component
+  mappingCmp: { [key: string]: ComponentType<any> };
+
+  labelMapper = (name: string) => this.entityCtr.schema.get(name).label;
+  isUsed = (option: string) =>
+    this.columnMapping.some(({ propertyName }) => propertyName === option);
+
+  constructor(
+    private entities: EntityRegistry,
+    private importService: ImportService,
+    private dialog: MatDialog
+  ) {}
+
+  ngOnChanges(changes: SimpleChanges): void {
+    if (changes.hasOwnProperty("entityType")) {
+      this.resetMappingToEntityType();
+    }
+  }
+
+  /**
+   * un-map properties not matching the entityType
+   * @private
+   */
+  private resetMappingToEntityType() {
+    this.columnMapping
+      .filter((c) => !this.allProps.includes(c.propertyName))
+      .forEach((c) => (c.propertyName = undefined));
+    this.updateMapping();
+  }
+
+  openMappingComponent(col: ColumnMapping) {
+    const uniqueValues = new Set();
+    this.rawData.forEach((obj) => uniqueValues.add(obj[col.column]));
+    this.dialog.open<any, MappingDialogData>(
+      this.mappingCmp[col.propertyName],
+      {
+        data: {
+          col: col,
+          values: [...uniqueValues],
+          entityType: this.entityCtr,
+        },
+        disableClose: true,
+      }
+    );
+  }
+
+  /**
+   * Emit an updated columnMapping array and emit change event, to ensure smooth change detection.
+   */
+  updateMapping() {
+    this.columnMappingChange.emit([...this.columnMapping]);
```

Comment:
for change detection to seamlessly work (update preview component, etc.), in place editing of elements is not sufficient. Therefore I wanted to avoid any side effect on the input value and instead emit a new array on each change. 

Admittedly, I only did that half-way because the data-binding in the template already edits the array ... If there is a better way to get change detection across components, we would consider that as well. Or clean up and on init create a copy, rather then here on emit?